substrate/frame/support/procedural/src/pallet/parse/config.rs

```diff
@@ -431,6 +475,15 @@ impl ConfigDef {
 			let mut already_no_default_bounds = false;
 			let mut already_collected_associated_type = None;
 
+			// add deprecation notice for `RuntimeEvent`, iff pallet is not `frame_system`
+			if is_event && has_frame_system_supertrait {
+				if let syn::TraitItem::Type(type_event) = trait_item {
+					type_event
+						.attrs
+						.push(syn::parse_quote!(#[deprecated(note = "`RuntimeEvent` associated type is deprecated, there is no need to define it in the pallet config.")]));
```

Comment:
We should use `proc_macro_warning` for issuing the warning to the pallet builder. Besides that we should provide an "allow(deprecated)" above the type to not print the warning.

Then we basically need to update all pallets in the repo to accept the deprecated `RuntimeEvent`. At some later point we then need to remove it. There is sadly no real better way for downstream as removing them at some point.